### PR TITLE
293 from 289 retrofit generate loop for logits

### PR DESCRIPTION
**Pull request by david-thrower**
Revised the generation to account for the change from probs to logits / switching to SparseCategoricalCrossentropy.

### CODE REVIEW

**Comment by david-thrower** [comment]
Merge into a dev branch.
